File: tools/conf/output/ts/ShopConfig.ts
class ShopConfig {
    private static _dic:Dictionary<ShopConfig> = new Dictionary<ShopConfig>();
    private static _arr:Array<ShopConfig> = new Array<ShopConfig>();

	 public Id :number ;
     public ItemId :number ;
     public ItemCount :number ;
    
     public Propaganda :string ;
     public Type :number ;
     public Tabid :number ;
     public Order :number ;
     public Cost :number ;
     public Price :number ;
     public PurchaseNums :number ;
     public Scale :number ;
     public Location : Array<number>  =new Array<number>();
     public NewStar :string ;
     public NewEnd :string ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<ShopConfig>{
        return  ShopConfig._dic;
    }
    public static get Arr():Array<ShopConfig>{
        return ShopConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:ShopConfig = <ShopConfig>json[i];
            ShopConfig._arr.push(config);
            ShopConfig._dic.add(config.Id,config);
        }
    }
}

File: server/src/game/conf/ShopConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type ShopConfigElem struct {
	Id int
	ItemId int
	ItemCount int
	Name string
	Propaganda string
	Type int
	Tabid int
	Order int
	Cost int
	Price int
	PurchaseNums int
	Scale float32
	Location []int
	NewStar string
	NewEnd string
	
}

type ShopConfig0 struct {
	Dic map[int]*ShopConfigElem
	Arr []*ShopConfigElem
}

var (
	ShopConfig ShopConfig0
)

func loadShopConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	ShopConfig.Dic = make(map[int]*ShopConfigElem)
	ShopConfig.Arr = make([]*ShopConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &ShopConfig.Arr)
	for _,conf := range ShopConfig.Arr{
		ShopConfig.Dic[conf.Id] = conf
	}
	
}
